pkg/addons/irsa_helper.go

```diff
@@ -0,0 +1,67 @@
+package addons
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	iamoidc "github.com/weaveworks/eksctl/pkg/iam/oidc"
+	"github.com/weaveworks/eksctl/pkg/kubernetes"
+	"github.com/weaveworks/eksctl/pkg/utils/tasks"
+)
+
+type serviceAccountCreator interface {
+	NewTasksToCreateIAMServiceAccounts(serviceAccounts []*api.ClusterIAMServiceAccount, oidc *iamoidc.OpenIDConnectManager, clientSetGetter kubernetes.ClientSetGetter, replaceExistingRole bool) *tasks.TaskTree
+}
```

Comment:
Can I ask why we have this little interface here?

It took me fractionally longer to parse the composition than if stackcollection (or stackmanager, whatever it is) were simply called from the helper struct on line 51.

Reply:
Because `irsaHelper` only needs this method and not the entire `StackCollection`. This makes testing the `VPCController` and `IRSAHelper` easier. It's somewhat similar to saying why, e.g, `ioutil.ReadAll` takes an `io.Reader` and not an `os.File` or  a `net.TCPConn`.

Reply:
Oh I couldn't find tests in this PR (I did assume that to start), are they still coming?

Reply:
I meant that as a general comment. I haven't added tests as the bulk of the logic is making Kubernetes and AWS calls.